NLSY/nlsy_impute.R

```diff
@@ -0,0 +1,160 @@
+#Packages
+librarian::shelf(dplyr, mice, car, corrplot, caret)
+
+# Read data
+nlsydf = readRDS(paste0(here::here(), "/NLSY/NLSY-college-finance.rds"))
+
+test <- nlsydf %>% #Add additional variables that should be considered during multiple imputation
+    select(id = "PUBID_1997",
+           mom_age_birth = "CV_BIO_MOM_AGE_YOUTH_1997", # numeric
+           age_resp =  "CV_AGE_12/31/96_1997",
+           hispanic = "KEY_ETHNICITY_1997", # binary
+           race = "KEY_RACE_1997", # categorical
+           mom_education = "CV_HGC_RES_MOM_1997", # categorical - highest grade completed in 1997 by mom
+           dad_education = "CV_HGC_RES_DAD_1997", # categorical - highest grade completed in 1997 by dad
+           savings = "P5-130_1997", #binary - yes/no retirement savings
+           home = "P5-101_1997", # binary - homeownership or renting
+           both_parents = "YOUTH_BOTHBIO.01_1997",
+           par1_deceased = "YOUTH_NONR1DEAD.01_1997",
+           par2_deceased = "YOUTH_NONR2DEAD.01_1997",
+           pincome = "CV_INCOME_GROSS_YR_1997",
+           pnetworth = "CV_HH_NET_WORTH_P_1997",
+           # Also, it seems that this represents the wealth of the household in which a respondent lives. That means that for a respondent who lives with only one parent, the net-worth variable represents the wealth of that parent.
+           # In the case of an independent respondent, contains respondent's, rather than parents', wealth.
+    ) %>%
+    mutate(mom_age = mom_age_birth + age_resp)
+
+# Making dummy variables
+test$hisp <- ifelse(test$hispanic == "Yes", 1, 0)
+test$ret_sav <- ifelse(test$savings == "YES", 1, 0)
+test$hown <- ifelse(test$home == "OWNS OR IS BUYING; LAND CONTRACT", 1, 0)
+test$both_pars <- ifelse(test$both_parents == "Yes", 1, 0)
+test$par_dec <- ifelse(test$par1_deceased == "Yes" | test$par2_deceased == "Yes", 1, 0) # Either parent deceased
+test$mom_educ_hs <- ifelse(test$mom_education == "12TH GRADE" | test$mom_education == "1ST YEAR COLLEGE" | test$mom_education == "2ND YEAR COLLEGE" |  test$mom_education == "3RD YEAR COLLEGE" |  test$mom_education == "4TH YEAR COLLEGE" |  test$mom_education == "5TH YEAR COLLEGE" |  test$mom_education == "6TH YEAR COLLEGE" |  test$mom_education == "7TH YEAR COLLEGE" |  test$mom_education == "8TH YEAR COLLEGE", 1, 0)
+test$dad_educ_hs <- ifelse(test$dad_education == "12TH GRADE" | test$dad_education == "1ST YEAR COLLEGE" | test$dad_education == "2ND YEAR COLLEGE" |  test$dad_education == "3RD YEAR COLLEGE" |  test$mom_education == "4TH YEAR COLLEGE" |  test$mom_education == "5TH YEAR COLLEGE" |  test$mom_education == "6TH YEAR COLLEGE" |  test$mom_education == "7TH YEAR COLLEGE" |  test$mom_education == "8TH YEAR COLLEGE", 1, 0)
+test$race_w <- ifelse(test$race == "White", 1, 0)
+test$race_b <- ifelse(test$race == "Black or African American", 1, 0)
```

Comment:
Similar to education categories, we should use more than just these two. We are not limited by the number of race categories in DYNASIM and sample size of a subgroup here does not play a role. For same reasons, I would leave ethnicity in a separate variable from race. For variables that are relevant, we want them to be as rich as possible.

Reply:
Agreed! I think just using all categories and creating a complete set of dummy variables from the original `race` variable would be good here!

Reply:
Great! Have deleted the race dummy and kept the original race variable as a factor.

Reply:
To close the loop on this - the interaction term and the race term were causing the mice package to report a singularity error (likely multicollinearity, which was difficult to pick up given they are both now factor variables). In the final imputation, I have only used the interaction term.